api/admin/controller.py

```diff
@@ -247,6 +251,28 @@ def generate_csrf_token(self):
         """Generate a random CSRF token."""
         return base64.b64encode(os.urandom(24))
 
+class AdminCirculationManagerController(CirculationManagerController):
+    """Parent class that provides methods for verifying an admin's roles."""
+
+    def require_system_admin(self):
+        admin = flask.request.admin
+        if not admin or not admin.is_system_admin():
+            raise AdminNotAuthorized()
```

Comment:
It might be better to use decorators for some of this, but some controllers have more complex logic around who can do things. I like having it at the controller-level rather than the route because it's easier to test.

Reply:
Could you decorate the controller methods rather than the routes? That would the non-complicated cases clear while keeping everything testable.

Reply:
It would also be easier to test, because in most cases just need to verify that the decorator was applied, you wouldn't actually have to set up a fake request.